### ISSUE TITLE
Webui Hangs after adding heavy workload to ipfs cluster peers

### ISSUE THREAD

**Issue by trendsetter37** [ipfs/ipfs-webui#2143]
**Description**
So I’ve come across a weird situation where my private cluster was empty, The only operations we used were just adding small test files to watch how they were distributed. During this time the webui for each private node continued to work as usual.

However, after adding a heavier load ~ 10GB of data to the private cluster using ips-cluster-ctl, the webui stopped working for each node. Good news is that the gateways still work but this seems like a weird bug. After the data was added to the private cluster the webui hangs if we attempt to navigate to the url in the browser.

**To Reproduce**

1. Install and start private ipfs nodes using swarm keys.
2. Check webui (still works at this point)
3. install connect nodes to ipfs-cluster service
4. Check webui (still works)
5. Load at least 10GB of data through the cluster
6. Check webui (Starts to hang when attempting to access)

**Expected behavior**
WebUI loads and is working normally.

**Desktop (please complete the following information):**
 - OS: Linux servers (Arch, Rasbian)
 - Browsers: Brave, Firefox, Palemoon
 - Version: Most recent

**Additional context**
It is interesting that the problem only occurs after loading more that a trivial amount of data to the cluster. Unsure of what the lower threshold would be.

**Comment by trendsetter37** [comment]
@whizzzkid Can you say more about regarding this: `I am not surprised that this happened` I can poke around the relevant code as well to look for clues. It sounds as if you already expected something like this to happen based on your knowledge of the relevant code surrounding webui/ipfs cluster stuff

**Comment by SgtPooki** [comment]
@trendsetter37 thanks for reporting this.

Can you confirm the following?

1. You added content to your ipfs node via CLI (i.e. outside of webui)
2. You attempted to load the main webui page and it hangs (127.0.0.1:5001/webui on a local Kubo node)

For #1, I want to make sure we're looking at the correct issue. For #2, if webui hangs on the initial load (regardless of page) this seems like it could be an issue exacerbated by the content on the node, or an ipfsd-ctl issue when attempting to connect to that node, and not a fully contained webui issue (i.e. some bug in our processing of large data). Though either way, we still have some work to do. 

If it's hanging on accessing a particular page, then that means some operations more specific to that page might be at fault.

I'm just trying to narrow down where our focus should be. 

Thanks for your continued input!

**Comment by trendsetter37** [comment]
@SgtPooki yes, all content was added via CLI and the main page is what never loads. I haven't attempted to load any secondary pages.

**Comment by SgtPooki** [comment]
Ok thanks 🙏. I figured but wanted to confirm

**Comment by trendsetter37** [comment]
Update here: I'm also getting a 302 status when attempting to go directly to `http://localhost:5001/webui`. Is that expected?

**Comment by trendsetter37** [comment]
@whizzzkid yes

**Comment by SgtPooki** [comment]
If you're running a kubo node, it does a temp redirect(302) from the /webui path to the appropriate cid for the webui

**Comment by gammazero** [comment]
There have been many changes since this issue was reported, and it is likely that this is already fixed. Please file a new issue is this or similar still exist.
